### PR TITLE
#63 - give warning if no contigs in common between bam/GFF

### PR DESCRIPTION
**Pull request by davmlaw**
New and improved PR

* Forked from latest master
* Allow edits by maintainer
* Added tests to ensure message was included in stderr when it should be, and not when it shouldn't

### CODE REVIEW

**Comment by davmlaw** [comment]
@iosonofabio I am not sure what is wrong with the build, appears to be git cloning issues (not setup right with pull requests?) you should be able to edit this time though

Let me know if you'd like me to do anything else, thanks.

File: example_data/no_contig_overlap_warning_stderr.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+2358 GFF lines processed.
+Alignment file '{FILENAME}' has no contigs in common with GFF/feature file. This will result in zero feature counts. Please check the references match, they can often vary eg using 'chr1' vs '1' as chromosome names
```
Comment (iosonofabio) [inline example_data/no_contig_overlap_warning_stderr.txt:2]: Why does it say `{FILENAME}` instead of the actual FILENAME?

**Comment by iosonofabio** [comment]
That's weird, don't worry about that, I'll fix it later if needed. Can you please address the comment and?

File: example_data/no_contig_overlap_warning_stderr.txt (hunk 1 above)
Comment (davmlaw) [reply example_data/no_contig_overlap_warning_stderr.txt:2]: The error message contains the full path of the file, this changes depending on where people check out the code and run the tests. 

I interpret the text file as a template, which I read and populate with the real path in test_no_contig_overlap_warning

```
        stderr_template = open(f'{data_folder}/no_contig_overlap_warning_stderr.txt').read()
        expected_stderr = stderr_template.format(FILENAME=bam_file)
```

**Comment by davmlaw** [comment]
@iosonofabio - I answered the question above (though it marked it as "resolved" and hid it - not sure if you saw it)

**Comment by iosonofabio** [comment]
Thanks for the reminder. I'll take a look asap, if you can push a dumb change to trigger a CI rerun that'd be great. It's hard to merge PRs that don't pass CI cause one risks to flatten the whole house of cards

**Comment by davmlaw** [comment]
I asked ChatGPT about the error and it suggested:

> Certainly! The error message is indicating that the remote branch with the name 65 was not found in the upstream repository. This likely happened because the command is trying to extract the branch name using cut -d/ -f3 on the GITHUB_REF environment variable, but it seems to be extracting a pull request number (in this case, 65) instead of the branch name.

> You may want to print the value of ```GITHUB_REF``` in your script to see what it contains. Typically, for a pull request, this variable will be something like refs/pull/65/merge, so the command you are using will extract 65 as the branch name.

A common way to clone the branch for a pull request is to use the following command:

```
git clone -b $GITHUB_HEAD_REF --single-branch https://github.com/$GITHUB_REPOSITORY.git
```

Here, GITHUB_HEAD_REF is the name of the branch where the changes are introduced. Make sure this environment variable is available in the context where your code is running.

Adjust your script accordingly, and this should fix the issue.

**Comment by iosonofabio** [comment]
Hmm, I maintain many git repos and this is the only instance of this error. I'll take a look.

I don't think asking chatGPT is the solution to the problem

**Comment by davmlaw** [comment]
OK good luck. I tried to modify it in my pull request but was rejected with:

```
! refs/heads/master:refs/heads/master [remote rejected] (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/ci.yml` without `workflow` scope)
```

**Comment by davmlaw** [comment]
I found a stack overflow answer -  https://stackoverflow.com/a/58035262

They explain that GITHUB_HEAD_REF is only for pull requests so maybe something like:

```
if [ "$GITHUB_EVENT_NAME" = "pull_request" ]; then
  BRANCH_NAME=$GITHUB_HEAD_REF
else
  BRANCH_NAME=$(echo ${GITHUB_REF} | cut -d/ -f3)
fi

git clone -b $BRANCH_NAME --single-branch https://github.com/${GITHUB_REPOSITORY}.git
```

**Comment by davmlaw** [comment]
There are a few solutions listed on that page - I think the easeist is to use the new variable:



GITHUB_REF_NAME - The short ref  name of the branch or tag that triggered the workflow run. This value  matches the branch or tag name shown on GitHub. For example, feature-branch-1.

**Comment by iosonofabio** [comment]
Thanks. I understand now. Not your fault, the CI script is just a little old that's all. I'll push a few things into the branch to try and get it to pass CI, then will merge.

**Comment by iosonofabio** [comment]
Merged, thank you!

**Pull request merged**